cdn/signUrl/SignUrlSample/Program.cs

```diff
@@ -0,0 +1,128 @@
+﻿// Copyright 2018 Google Inc. All rights reserved.
+// Use of this source code is governed by the Apache 2.0
+// license that can be found in the LICENSE file.
+
+using CommandLine;
+using System;
+using System.IO;
+using System.Linq;
+using System.Security.Cryptography;
+using System.Text;
+
+namespace SignUrlSample
+{
+    [Verb("signurls")]
+    class SignedUrlArgs
+    {
+        [Value(0, HelpText = "Url to sign", Required = true)]
+        public string Url { get; set; }
+
+        [Option('k', "key-name",
+            HelpText = "The name of the key to use when encrypting url")]
+        public string KeyName { get; set; }
+
+        [Option('p', "key-path",
+            HelpText = "The path to the key to use when encrypting url")]
+        public string KeyPath { get; set; }
+
+        [Option('e', "expiration",
+            HelpText = "Url expiration in UTC formatted as: YYYY-MM-DDThh:mm:ss")]
+        public DateTime Expiration { get; set; }
+    }
+
+    public class Program
+    {
+        static int Main(string[] args)
+        {
+            string signedUrl = null;
+            Parser.Default.ParseArguments<SignedUrlArgs>(args).MapResult(
+                uargs =>
+                    signedUrl = CreateSignedUrl(
+                                    uargs.Url,
+                                    uargs.KeyName,
+                                    File.ReadAllText(uargs.KeyPath, Encoding.UTF8),
+                                    uargs.Expiration),
+                err => string.Empty);
+
+            if (string.IsNullOrEmpty(signedUrl))
+            {
+                return 1;
+            }
+
+            Console.WriteLine($"Signed URL: {signedUrl}");
+            return 0;
+        }
+
+        // [START CreateSignedUrl]
+        /// <summary>
+        /// Creates signed URL for Google Cloud SDN
+        /// More details about order of operations is here: 
+        /// <see cref="https://cloud-dot-devsite.googleplex.com/cdn/docs/signed-urls#creatingkeys"/>
+        /// </summary>
+        /// <param name="url">The Url to sign. This URL can't include Expires and KeyName query parameters in it</param>
+        /// <param name="keyName">The name of the key used to sign the URL</param>
+        /// <param name="encodedKey">The key used to sign the Url</param>
+        /// <param name="expirationTime">Expiration time of the signature</param>
+        /// <returns>Signed Url that is valid until {expirationTime}</returns>
+        public static string CreateSignedUrl(string url, string keyName, string encodedKey, DateTime expirationTime)
+        {
+            var builder = new UriBuilder(url);
+
+            long unixTimestampExpiration = ToUnixTime(expirationTime);
+
+            char queryParam = string.IsNullOrEmpty(builder.Query) ? '?' : '&';
+            builder.Query += $"{queryParam}Expires={unixTimestampExpiration}&KeyName={keyName}".ToString();
```

Comment:
Do we need to `Uri.EscapeDataString()` the `keyName`?

Reply:
No, we asked the product team and they told the values for this param is url safe.